In [1]:
import numpy as np 
import pandas as pd

import matplotlib.pyplot as plt
import seaborn as sns
from subprocess import check_output

In [2]:
# Se carga el dataset

Dataset = "../input/fivethirtyeight/data/data/nutrition-studies/nutrition-studies/raw_anonymized_data.csv"
d = pd.read_csv(Dataset)

# **1 - Descripcion del data set**

### **Resumen del DataSet**

In [3]:
# Vemos las primeras y las ultimas 10 filas.

the_first_teen = d.head(10)
the_last_teen  = d.tail(10)

reduced_dataset = pd.concat([the_first_teen, the_last_teen], axis=0) # Apilamos uno encima del otro

reduced_dataset.head(20)

ID cancer diabetes heart_disease  belly ever_smoked currently_smoke  \
0   1003    Yes       No            No  Innie         Yes             Yes   
1   1053     No      Yes           Yes  Outie         Yes             Yes   
2   1006    Yes      Yes           Yes  Innie          No              No   
3   1166     No       No            No  Innie          No              No   
4   1134    Yes       No            No  Innie          No              No   
5   1014     No       No            No  Innie         Yes             Yes   
6   1074    Yes       No            No  Innie          No              No   
7   1151    Yes       No           Yes  Innie          No              No   
8   1001    Yes      Yes           Yes  Innie         Yes              No   
9   1048    Yes       No            No  Innie          No              No   
44  1153     No       No            No  Innie          No              No   
45  1128    Yes       No           Yes  Innie         Yes              No   
46  1002     No       No            No  Innie          No              No   
47  1026    Yes      Yes           Yes  Innie          No              No   
48  1013    Yes       No            No  Innie          No              No   
49  1129    Yes       No            No  Innie          No              No   
50  1005    Yes      Yes            No  Innie          No              No   
51  1044    Yes       No            No  Innie          No              No   
52  1045    Yes       No           Yes  Innie         Yes              No   
53  1093     No       No            No  Innie          No              No   

   smoke_often smoke_rarely never_smoked  ... DT_FIBER_INSOL DT_FIBER_SOL  \
0          Yes           No           No  ...           7.38         1.25   
1           No          Yes           No  ...           9.11         3.37   
2           No           No          Yes  ...          11.56         4.74   
3           No           No          Yes  ...          26.34        10.85   
4           No           No          Yes  ...          16.48         4.80   
5           No          Yes           No  ...          20.32         8.55   
6           No           No          Yes  ...          10.79         3.28   
7           No           No          Yes  ...          10.31         4.63   
8           No           No           No  ...           9.96         3.81   
9           No           No          Yes  ...          20.54         8.18   
44          No           No          Yes  ...          19.33         7.13   
45          No           No           No  ...          20.13         7.96   
46          No           No          Yes  ...          30.75        11.35   
47          No           No          Yes  ...          37.72        13.93   
48          No           No          Yes  ...           7.96         3.34   
49          No           No          Yes  ...          20.14         5.55   
50          No           No          Yes  ...          11.11         3.67   
51          No           No          Yes  ...          15.49         6.10   
52          No           No           No  ...          28.38         8.68   
53          No           No          Yes  ...          17.93         8.34   

   DT_PROT_ANIMAL DT_PROT_VEGETABLE DT_NITROGEN PHYTIC_ACID OXALIC_ACID  \
0           75.46             16.00       14.89      365.70      318.11   
1           59.41             18.25       12.51      434.98      112.66   
2           61.49             28.46       14.45      606.43      213.41   
3           28.71             44.59       12.15     1570.07      334.08   
4           32.41             28.23        9.80      616.99      422.55   
5           13.53             32.42        7.51      850.24      186.17   
6           11.69             25.11        6.12      550.18      125.83   
7           34.37             23.26        9.28      503.17      135.44   
8           19.87             22.68        6.92      448.21      331.86   
9           55.55   

### **Dimensiones**

In [4]:
# Se muestra dimensiones: (filas, columnas)
d.shape

(54, 1093)

> **Cantidad de Filas: 54**

> **Cantidad de Columnas: 1093**

### **Cantidad de datos Nulos**

In [5]:
# Contar cuantas columnas contiene ALGÚN dato nulo

contador_nulls = 0

for columna in d:  
    if(d[columna].isnull().any().sum() > 0): 
        contador_nulls = contador_nulls + 1 
    
print(contador_nulls)


0


In [6]:
# Contar cuantas filas contiene ALGÚN dato nulo

contador_nulls = 0

for i in range(0,54):
    if(d.iloc[i].isnull().any().sum() > 0):
        contador_nulls = contador_nulls + 1 
    
print(contador_nulls)

0


> **No hay datos faltantes en ninguna fila**

> **No hay datos faltantes en ninguna columna**

### **Tipos de datos**

In [7]:
# Analizo tipos de datos
d.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 54 entries, 0 to 53
Columns: 1093 entries, ID to FORMONONETIN
dtypes: float64(554), int64(513), object(26)
memory usage: 461.2+ KB


**El dataset posee tres tipos de datos:**

> 554 de tipo float

> 513 de tipo int

> 26 de tipo object


In [8]:
d.select_dtypes(include = 'int64').head()

ID  BREAKFASTSANDWICHFREQ  BREAKFASTSANDWICHQUAN  EGGSFREQ  EGGSQUAN  \
0  1003                      5                      1         7         3   
1  1053                      1                      1         4         4   
2  1006                      3                      2         4         2   
3  1166                      2                      1         7         2   
4  1134                      3                      1         5         2   

   YOGURTFREQ  YOGURTQUAN  COTTAGECHEESEFREQ  COTTAGECHEESEQUAN  \
0           1           3                  1                  2   
1           2           3                  1                  2   
2           5           2                  2                  2   
3           8           3                  1                  2   
4           8           2                  2                  2   

   CREAMCHEESEFREQ  ...  GROUP_CREAM_OR_HALF_N_HALF_TOTAL_GRAMS  \
0                7  ...                                       0   
1                2  ...                                       0   
2                6  ...                                       0   
3                1  ...                                       0   
4                4  ...                                       0   

   GROUP_NON_DAIRY_CREAMER_LIQUID_TOTAL_GRAMS  \
0                                           0   
1                                           0   
2                                           0   
3                                           0   
4                                           0   

   GROUP_CONDENSED_MILK_TOTAL_GRAMS  GROUP_SUGAR_OR_HONEY_TOTAL_GRAMS  \
0                                 0                                 0   
1                                 0                                 0   
2                                 0                                 0   
3                                 0                                 0   
4                                 0                                 0   

   GROUP_COOKING_FAT_POP_MIX_TOTAL_GRAMS  \
0                                      0   
1                                      0   
2                                      0   
3                                      0   
4                                      0   

   GROUP_NON_STICK_SPRAY_SR27_TOTAL_GRAMS  \
0                                       0   
1                                       0   
2                                       0   
3                                       0   
4                                       0   

   GROUP_COOK_FAT_PEANUT_OIL_TOTAL_GRAMS  \
0                                      0   
1                                      0   
2                                      0   
3                                      0   
4                                      0   

   GROUP_COOK_FAT_VEG_SHORTENING_CRISCO_TOTAL_GRAMS  F20D4N6  F21D5  
0                                                 0        0      0  
1                                                 0        0      0  
2                                                 0        0      0  
3                                                 0        0      0  
4                                                 0        0      0  

[5 rows x 513 columns]

In [9]:
d.select_dtypes(include = 'float64').head()

GROUP_SOLID_TOTAL_FREQUENCY  GROUP_SOLID_TOTAL_GRAMS  \
0                         6.38                   667.04   
1                         4.78                   801.63   
2                         8.23                   855.93   
3                        12.76                  1633.78   
4                         7.10                   813.23   

   GROUP_ALCOHOLICBEVERAGES_TOTAL_KCAL  GROUP_ALCOHOLICBEVERAGES_TOTAL_GRAMS  \
0                               327.13                                413.05   
1                               141.42                                200.99   
2                                98.82                                218.65   
3                                 0.00                                  0.00   
4                               383.97                                542.44   

   GROUP_ALCOHOLICBEVERAGES_TOTAL_FREQUENCY  GROUP_SUGARYBEVG_TOTAL_KCAL  \
0                                     0.483                         0.00   
1                                     0.319                       207.81   
2                                     0.352                       141.39   
3                                     0.000                         0.00   
4                                     1.510                        29.22   

   GROUP_SUGARYBEVG_TOTAL_GRAMS  GROUP_SUGARYBEVG_TOTAL_FREQUENCY  \
0                          0.00                             0.000   
1                        569.86                             0.543   
2                        542.84                             0.555   
3                          0.00                             0.000   
4                        103.17                             0.110   

   GROUP_A_SUGBEV_TOTAL_KCAL  GROUP_A_SUGBEV_TOTAL_FREQUENCY  ...  \
0                       1.96                          0.0165  ...   
1                     215.72                          0.5930  ...   
2                     178.11                          0.6490  ...   
3                       0.00                          0.0000  ...   
4                     108.19                          0.4460  ...   

   DT_FIBER_INSOL  DT_FIBER_SOL  DT_PROT_ANIMAL  DT_PROT_VEGETABLE  \
0            7.38          1.25           75.46              16.00   
1            9.11          3.37           59.41              18.25   
2           11.56          4.74           61.49              28.46   
3           26.34         10.85           28.71              44.59   
4           16.48          4.80           32.41              28.23   

   DT_NITROGEN  PHYTIC_ACID  OXALIC_ACID  COUMESTROL  BIOCHANIN_A  \
0        14.89       365.70       318.11      0.0117       0.0658   
1        12.51       434.98       112.66      0.0107       0.1390   
2        14.45       606.43       213.41      0.0965       0.0519   
3        12.15      1570.07       334.08      0.2830       0.0890   
4         9.80       616.99       422.55      0.1630       0.0994   

   FORMONONETIN  
0       0.00324  
1       0.00743  
2       0.00946  
3       0.01260  
4       0.02070  

[5 rows x 554 columns]

In [10]:
d.select_dtypes(include = 'object').head()

cancer diabetes heart_disease  belly ever_smoked currently_smoke  \
0    Yes       No            No  Innie         Yes             Yes   
1     No      Yes           Yes  Outie         Yes             Yes   
2    Yes      Yes           Yes  Innie          No              No   
3     No       No            No  Innie          No              No   
4    Yes       No            No  Innie          No              No   

  smoke_often smoke_rarely never_smoked quit_smoking  ... neutralCable  \
0         Yes           No           No           No  ...           No   
1          No          Yes           No           No  ...           No   
2          No           No          Yes           No  ...          Yes   
3          No           No          Yes           No  ...           No   
4          No           No          Yes           No  ...          Yes   

  noCrash yesCrash uhCrash rash cat  dog Dems atheist Jewish  
0     Yes       No      No  Yes  No   No  Yes     Yes     No  
1      No       No     Yes  Yes  No   No  Yes     Yes     No  
2      No       No     Yes   No  No  Yes   No      No     No  
3      No       No     Yes   No  No   No  Yes     Yes     No  
4      No       No     Yes   No  No   No  Yes     Yes     No  

[5 rows x 26 columns]

## **2 - Análisis de Variables Cualitativas**

In [11]:
# Hago una copia de las columnas con variables cualitativas
d_copy = d.select_dtypes(include = 'object')

In [12]:
d_copy.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 54 entries, 0 to 53
Data columns (total 26 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   cancer           54 non-null     object
 1   diabetes         54 non-null     object
 2   heart_disease    54 non-null     object
 3   belly            54 non-null     object
 4   ever_smoked      54 non-null     object
 5   currently_smoke  54 non-null     object
 6   smoke_often      54 non-null     object
 7   smoke_rarely     54 non-null     object
 8   never_smoked     54 non-null     object
 9   quit_smoking     54 non-null     object
 10  left_hand        54 non-null     object
 11  right_hand       54 non-null     object
 12  readingMath      54 non-null     object
 13  mathReading      54 non-null     object
 14  favCable         54 non-null     object
 15  unfavCable       54 non-null     object
 16  neutralCable     54 non-null     object
 17  noCrash          54 non-null     obje

In [13]:
d_copy.head()

cancer diabetes heart_disease  belly ever_smoked currently_smoke  \
0    Yes       No            No  Innie         Yes             Yes   
1     No      Yes           Yes  Outie         Yes             Yes   
2    Yes      Yes           Yes  Innie          No              No   
3     No       No            No  Innie          No              No   
4    Yes       No            No  Innie          No              No   

  smoke_often smoke_rarely never_smoked quit_smoking  ... neutralCable  \
0         Yes           No           No           No  ...           No   
1          No          Yes           No           No  ...           No   
2          No           No          Yes           No  ...          Yes   
3          No           No          Yes           No  ...           No   
4          No           No          Yes           No  ...          Yes   

  noCrash yesCrash uhCrash rash cat  dog Dems atheist Jewish  
0     Yes       No      No  Yes  No   No  Yes     Yes     No  
1      No       No     Yes  Yes  No   No  Yes     Yes     No  
2      No       No     Yes   No  No  Yes   No      No     No  
3      No       No     Yes   No  No   No  Yes     Yes     No  
4      No       No     Yes   No  No   No  Yes     Yes     No  

[5 rows x 26 columns]

## **3 - Análisis de Variables Cuantitativas**

In [14]:
#Holus! This is Bren's branch
